ksqldb-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java

```diff
@@ -115,17 +115,18 @@ public SchemaRegistryClient get() {
     }
   
     final RestService restService = serviceSupplier.get();
-    if (sslContext != null) {
-      restService.setSslSocketFactory(sslContext.getSocketFactory());
-    }
-
-    return schemaRegistryClientFactory.create(
+    final SchemaRegistryClient client = schemaRegistryClientFactory.create(
         restService,
         1000,
         ImmutableList.of(
             new AvroSchemaProvider(), new ProtobufSchemaProvider(), new JsonSchemaProvider()),
         schemaRegistryClientConfigs,
         httpHeaders
     );
+    // Let's reset the sslContext...
```

Comment:
just for my understanding, what happens if this sslContext is not reset?

Reply:
The code above `schemaRegistryClientFactory.create(...)` calls https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java#L181-L232

The last lines of that method call `restService.setSslSocketFactory(...)`...  effectively, that undoes the code that was setting the factory before we called the constructor.